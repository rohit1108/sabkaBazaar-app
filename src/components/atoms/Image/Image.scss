@import "../../../styles/variables";

@mixin image-mixin($width-max, $height-max) {
  max-width: $width-max;
  max-height: $height-max;
}

// Default Class for Images
.image-wrapper {
  width: 100%;
  height: 100%;
  @include image-mixin(200px, 150px);
  object-fit: contain;

  @media screen and (min-width: $mobile-breakpoint) {
    @include image-mixin(160px, 120px);
  }
}
