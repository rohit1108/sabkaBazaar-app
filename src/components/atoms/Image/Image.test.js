import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Image from './Image';
describe("Image component", () => {
    it('renders the component properly', () => {
        render(<Image />);
    });

    it('should receive props correctly', () => {
        render(<Image source={"./static/images/products/fruit-n-veg/kiwi-green.jpg"} 
                      alt={"Fresho Kiwi - Green, 3 pcs"}
                      className={"Image-1"}/>);
        screen.getByAltText("Fresho Kiwi - Green, 3 pcs");

    });

    it('should have the right Image component snapshot', () => {
        const tree = renderer.create(<Image />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});