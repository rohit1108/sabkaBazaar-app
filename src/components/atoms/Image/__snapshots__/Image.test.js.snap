// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component should have the right Image component snapshot 1`] = `
<img
  className=" image-wrapper"
  data-testid="image"
/>
`;

exports[`renders correctly across screens 1`] = `
<img
  className=" image-wrapper"
/>
`;
